### PR TITLE
[Feature] Add Documentation Section

### PR DESCRIPTION
**Pull request by kylecesmat**
The documentation section lives on its own route at `/docs`. There, we will have a sidebar with the various sections/guides. This menu is based on `radium/docs/` and the default view/item is `radium/readme`. The files are interpreted by ecology.

The layout is responsive, and the navigation will move off-canvas for smaller-screens.

Preview:

### CODE REVIEW

**Comment by kylecesmat** [comment]
Okay, this should be ready for review @paulathevalley

File: src/views/docs/components/docs.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        </header>
+        <section
+          style={styles.documentContainer}
+          ref="docContainer"
+          className="DocumentContainer"
+        >
+          <Ecology
+            overview={this.props.docs}
+            scope={{React, ReactDOM, Radium}}
+            playgroundtheme="elegant"
```
Comment (paulathevalley) [inline src/views/docs/components/docs.js]: I think there may be two code highlighting libraries here. `Ecology` uses `react-code-mirror`, which will apply the "elegant" theme as the `.cm-s-elegant` class to `<div class="CodeMirror">`. 

For builder and spectacle, there were no component playgrounds, so `ecology` was removed entirely. Instead, builder and spectacle use `marked` to parse the `README.md` directly and the prism.js library to highlight the code snippets, which requires that fun `Prism.highlightAll();` and importing the appropriate code libraries. 

Both use different classes for theme, so it should be easy to see which/both are happening.

Comment (kylecesmat) [inline src/views/docs/components/docs.js]: @paulathevalley Yes, there are. Maybe I need to revisit Ecology but my understanding was that Ecology only rendered playgrounds with previews? Not codeblocks with syntax-highlighting.

I'll revisit this and look at how I can only use one library.

**Comment by kylecesmat** [comment]
@paulathevalley should be ready for re-review now!

File: src/views/docs/components/docs.js (hunk 1 above)
Comment (paulathevalley) [inline src/views/docs/components/docs.js]: Ah, sorry I am being unclear! Did you figure it out?

Ecology renders Component Playgrounds, but Playgrounds render Code Mirror code blocks. Code Mirror will apply syntax highlighting with the `.cm-s-elegant` theme, you can see the stylesheet here: https://github.com/FormidableLabs/radium-docs/blob/bcf4fcaffd8b45834ecfa174f9f69a5c715d74d9/index.html#L12 

Here is the Code Mirror syntax highlighting for victory: https://github.com/FormidableLabs/formidable-landers/blob/master/src/themes/victory.js#L519-L601

File: src/radium-stylesheet.js [hunk 2]
```diff
@@ -8,6 +8,20 @@ import theme from "./radium-theme";
...
+const codeStyles = {
+  fontFamily: theme.code,
+  fontSize: ".9em",
+  lineHeight: 1.375,
+  direction: "ltr",
+  textAlign: "left",
+  whiteSpace: "pre",
+  wordSpacing: "normal",
+  wordBreak: "normal",
+  tabSize: 4,
```
Comment (paulathevalley) [inline src/radium-stylesheet.js]: whoa, we can control the indentation space? This is so cool!

Comment (kylecesmat) [inline src/radium-stylesheet.js]: yeah! Assuming that the space in question is indeed a tab! https://developer.mozilla.org/en-US/docs/Web/CSS/tab-size

**Comment by paulathevalley** [comment]
I am enlightened now. :+1:

**Pull request merged**